src/app/features/owner-restaurants/owner-restaurants.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { AddingModalComponent } from 'src/app/shared/adding-modal/adding-modal.component';
+import { OwnerRestaurantsService } from '../../core/services/owner-restaurants.service';
+import { Subscription } from 'rxjs';
+import { Restaurant } from 'src/app/core/interfaces/restaurant';
+
+@Component({
+  selector: 'owner-restaurants',
+  templateUrl: './owner-restaurants.component.html',
+  styleUrls: ['./owner-restaurants.component.scss'],
+})
+export class OwnerRestaurantsComponent implements OnInit {
+  restaurants: Restaurant[] = [];
+  ownerRestaurantSubscription: Subscription;
```

Comment:
Also, unsubscribe from this in the ngOnDestroy? Read about why we have to unsubscribe in ngOnDestroy